domain/containerimageresourcestore/service/service.go

```diff
@@ -9,23 +9,47 @@ import (
 	"encoding/json"
 	"io"
 
-	"github.com/juju/juju/domain/application/resource"
-	charmresource "github.com/juju/juju/internal/charm/resource"
+	"github.com/juju/juju/core/resource/store"
+	"github.com/juju/juju/domain/containerimageresourcestore"
 	"github.com/juju/juju/internal/docker"
 	"github.com/juju/juju/internal/errors"
 )
 
-func newContainerImageResourceStore(st ContainerImageMetadataState) *containerImageResourceStore {
-	return &containerImageResourceStore{st: st}
+// State provides methods for interacting
+// with the container image resource store.
+type State interface {
+	// RemoveContainerImageMetadata removes a container image resources metadata
+	// from the container image metadata resource store.
+	RemoveContainerImageMetadata(
```

Comment:
These method names can be changed to `Remove`, `Put` and `Get` now they are on their own state.

Reply:
They should always have a suffix. It was wrong not to do that in other domains.